### PR TITLE
Add bulk file metadata updates

### PR DESCRIPTION
**Pull request by MancunianSam**
This is part of the change to the backend checks architecture. We're now going to be updating the API once all the files are processed. I'm going to keep the FFID, AV and Checksum updates separate because I don't want to overload the database too much.
I've left the original addFFIDMetadata in so the existing FFID checks will still work but once they're moved over, we can delete it.
I've updated all the tests to point to the new endpoint.

### CODE REVIEW

File: src/main/scala/uk/gov/nationalarchives/tdr/api/service/FFIDMetadataService.scala [hunk 1]
```diff
@@ -27,37 +26,42 @@ class FFIDMetadataService(
...
-      uuidSource.uuid,
-      ffidMetadata.fileId,
-      ffidMetadata.software,
-      ffidMetadata.softwareVersion,
-      Timestamp.from(timeSource.now),
-      ffidMetadata.binarySignatureFileVersion,
-      ffidMetadata.containerSignatureFileVersion,
-      ffidMetadata.method
-    )
+  @deprecated("Use addFFIDMetadata(ffidMetadataInput: FFIDMetadataInput)")
```
Comment (TomJKing) [inline src/main/scala/uk/gov/nationalarchives/tdr/api/service/FFIDMetadataService.scala:29]: Nice.

File: src/main/scala/uk/gov/nationalarchives/tdr/api/service/FFIDMetadataService.scala [hunk 2]
```diff
@@ -27,37 +26,42 @@ class FFIDMetadataService(
...
+      })
+      .unzip
 
-    (for {
-      fileStatusRows <- generateFileStatusRows(ffidMetadata)
-      _ = loggingUtils.logFileFormatStatus("FFID", ffidMetadata.fileId, fileStatusRows.map(_.value).mkString(","))
-      ffidMetadataRow <- ffidMetadataRepository.addFFIDMetadata(metadataRow, fileStatusRows)
-      ffidMetadataMatchesRow <- addFFIDMetadataMatches(ffidMetadataRow.ffidmetadataid)
+    for {
+      statuses <- Future.sequence(ffidMetadataInput.metadataInputValues.map(generateFileStatusRows))
```
Comment (TomJKing) [inline src/main/scala/uk/gov/nationalarchives/tdr/api/service/FFIDMetadataService.scala:59]: Is there a reason that we're no longer logging the statuses?

**Review by TomJKing — COMMENTED** [review]
Looks good. Just a couple of very minor comments. Thanks

For my own sanity this won't be a breaking change will it?

File: src/main/scala/uk/gov/nationalarchives/tdr/api/service/FFIDMetadataService.scala (hunk 2 above)
Comment (MancunianSam) [reply src/main/scala/uk/gov/nationalarchives/tdr/api/service/FFIDMetadataService.scala:59]: I think I had some vague idea about replacing the Grafana monitoring that runs off these logs with something in the step functions but that's way off so I've put the logging back.

**Review by TomJKing — APPROVED** [review]
Looks good. Thanks

**Pull request merged**